HW1/P2/P2.txt

```diff
@@ -1 +1,2 @@
 # Your discussion here
+The histograms show a significant improvement of work balance among partitions. The computation can complete faster because the balanced work loading reduced the running time of partition needs the longest time. However, the repartition needs extra efforts to partition the tasks evenly. The trade off is this overhead for the repartition. This overhead needs to be shorter than the improvement of running time to worth doing the repartition.
```

Comment:
I think that it is important to think about a scenario when shuffling would slow down the computation. This would be the case when for example the list is very large and we at the same time have many processors and therefore divide the data in many small partitions. In this case the small partitions would be processed relatively quickly, but shuffling the list/RDD would take a relatively long time. It such case it might be better not to shuffle.